lib/console/deployments/observability/webhook/datadog.ex

```diff
@@ -0,0 +1,180 @@
+defmodule Console.Deployments.Observability.Webhook.Datadog do
+  @behaviour Console.Deployments.Observability.Webhook
+  import Console.Deployments.Observability.Webhook.Base
+
+  # Regular expressions to extract information from unstructured text
+  @service_regex [~r/service[:\s]+([^,\n]+)/i, ~r/service_name[:\s]+([^,\n]+)/i]
+  @cluster_regex [~r/cluster[:\s]+([^,\n]+)/i, ~r/cluster_name[:\s]+([^,\n]+)/i]
+  @project_regex [~r/project[:\s]+([^,\n]+)/i, ~r/namespace[:\s]+([^,\n]+)/i]
+
+  # Basic associations handlers
+  def associations(:project, %{"tags" => tags}, acc) when is_list(tags) do
```

Comment:
convert the tags list to a map first, then match the keys in the map internally in a helper function (will avoid the need to keep looping over that list compounding a ton of o(n) work)